crawl-ref/source/ng-wanderer.cc

```diff
@@ -536,6 +636,10 @@ static vector<spell_type> _wanderer_good_equipment(skill_type & skill)
     case SK_EARTH_MAGIC:
     case SK_POISON_MAGIC:
     case SK_HEXES:
+        if (one_chance_in(3))
+            return _give_wanderer_major_spells(skill);
+    // deliberate fallthrough
```

Comment:
Let's label this `// deliberate fallthrough to SK_SPELLCASTING`. Unlabeled fallthroughs are [dangerous](https://nethackwiki.com/wiki/Yeenoghu#.22A_ludicrous_bug.22) :)